microovn/frr/bgp/redirect.go

```diff
@@ -112,6 +112,19 @@ func generateLrpMac(ifaceName string) string {
 	return strings.TrimRight(macAddr, ":")
 }
 
+// generateRouterID returns a router-id address based on a string. The returned
+// router-id will always be same for given interface name.
+// Warning: There is no guarantee that the address won't conflict with other
+// router-ids present in the AS.
+func generateRouterID(s string) string {
```

Comment:
Function name and/or the documentation could clarify that this is pertains to BGP router ID. Not to be confused with, for example, OVN's Logical Router's ID.